site/app/templates/submission/homework/SubmitBox.twig

```diff
@@ -418,6 +420,69 @@
             }
             e.stopPropagation();
         });
+
+        var serverDate;
+
+        var fetchTimeUrl = window.location.href + "&action=fetch_time";
+
+        var nextTimeFetch = 0 ;
+        var setTimerOn = 0;
+
+        fetchServerTime();
+
+
+        function fetchServerTime() {
+
+            var localDate = new Date();
+
+            console.log("Fetching Server");
+            if(nextTimeFetch == 0) {
+                // console.log("Fetch set time remaining: "+(60 - localDate.getSeconds()));
+                nextTimeFetch = 300;
+                setTimeout(function(){
+                    setInterval(fetchServerTime, nextTimeFetch * 1000);
+                    // console.log("Next fetch set");
+                }, (60 - localDate.getSeconds())*1000);
+            }
+
+            $.ajax({
+                url: fetchTimeUrl,
+                dataType: "json",
+                success: function(result){
+                    localDate = new Date();
+                    serverDate = new Date((result.data.time + (localDate.getSeconds())) * 1000);
+                    console.log("Updated servertime");
+                    if(setTimerOn == 0){
+                        // console.log("Server time remaining: "+(60 - serverDate.getSeconds()));
+                        setTimeout(function(){
+                            // console.log("Setting interval");
+                            serverDate = new Date((result.data.time + (60 - serverDate.getSeconds()))* 1000);
```

Comment:
Why is there another nest setTimeout function when theres already one above the ajax call. If the ajax response is delayed for whatever reason I don't think this will correct time skew, it might make it worse if the ajax response is really out of sync and just overwrites the setTime function with an outdated time.